git_machete/cmd.py

```diff
@@ -1732,7 +1791,7 @@ def get_fork_point_override_data(b):
     return to_sha, while_descendant_of_sha
 
 
-def get_overridden_fork_point(b):
+def get_overridden_fork_point(b: str) -> Optional[str]:
```

Comment:
Btw, does mypy warn about potentially None values properly (like Kotlin/Checker Framework does)? or does it give a free pass on dereferencing None values (like Java/Scala)?

Reply:
https://mypy.readthedocs.io/en/stable/command_line.html#none-and-optional-handling

> `--no-implicit-optional`
This flag causes mypy to stop treating arguments with a None default value as having an implicit Optional type.
>
> For example, by default mypy will assume that the x parameter is of type Optional[int] in the code snippet below since the default parameter is None:
> ```
> def foo(x: int = None) -> None:
>     print(x)
> ```
>If this flag is set, the above snippet will no longer type check: we must now explicitly indicate that the type is Optional[int]:
>
> ```
> def foo(x: Optional[int] = None) -> None:
>     print(x)
> ```

Reply:
whoops, N/A 
it is already used

Reply:
however turning `stirct-optional` on results with about 70 errors like that:
```
➜  git-machete git:(refactor/mypy) ✗ mypy --config-file mypy.ini git_machete
git_machete/cmd.py:355: error: Argument 1 to "len" has incompatible type "Optional[str]"; expected "Sized"
git_machete/cmd.py:356: error: Unsupported operand types for * ("None" and "int")
git_machete/cmd.py:356: note: Left operand is of type "Optional[str]"
git_machete/cmd.py:359: error: Item "None" of "Optional[str]" has no attribute "__iter__" (not iterable)
git_machete/cmd.py:430: error: Unsupported operand types for * ("int" and "None")
git_machete/cmd.py:430: note: Right operand is of type "Optional[str]"
...
```

Reply:
In particular, this is the warning you asked for 
```
git_machete/cmd.py:1775: error: Argument 1 to "commit_sha_by_revision" has incompatible type "Optional[str]"; expected "str"
```

Reply:
Whoops okay so pls add an issue for enabling strict-optional one day

Reply:
created #118